File: MasterLogicaProgramacion/EjerciciosPython/Ej6.py
def hacerCuadrado(tamano):
    if tamano == 1:
        return "*"

    lineaDentro = "*" + " " * (tamano - 2) + "*\n"
    lineaFuera = "*" * tamano + "\n"
    cuadrado = lineaFuera

    for i in range(0, tamano - 2):
        cuadrado += lineaDentro

    cuadrado += lineaFuera

    return cuadrado

def hacerCuadradoBucles(tamano):
    if (tamano == 1):
        return "*"
    
    j = 0
    lineaDentro = "*"
    lineaFuera = "*"

    while (j < tamano - 2):
        lineaDentro += " "
        lineaFuera += "*"

        j+= 1

    lineaDentro += "*\n"
    lineaFuera += "*\n"
    
    cuadrado = lineaFuera

    for i in range(tamano - 2):
        cuadrado += lineaDentro

    cuadrado += lineaFuera

    return cuadrado

print(hacerCuadrado(4))
print(hacerCuadradoBucles(4))

File: Cursos/MasterLogicaProgramacion/EjerciciosJavascript/Ej6.js
function hacerCuadrado(tamano) {
    if (tamano == 1) return "*";

    let lineaDentro = "*" + " ".repeat(tamano - 2) + "*\n";
    let lineaFuera = "*".repeat(tamano) + "\n";
    let cuadrado = lineaFuera;

    for (let i = 0; i < (tamano - 2); i++) {
        cuadrado += lineaDentro;
    }

    cuadrado += lineaFuera;

    return cuadrado;
}

function hacerCuadradoBucles(tamano) {
    if (tamano == 1) return "*";
    
    let j = 0;
    let lineaDentro = "*";
    let lineaFuera = "*";

    while (j < tamano - 2) {
        lineaDentro += " ";
        lineaFuera += "*";

        j++;
    }

    lineaDentro += "*\n";
    lineaFuera += "*\n";
    
    let cuadrado = lineaFuera;

    for (let i = 0; i < (tamano - 2); i++) {
        cuadrado += lineaDentro;
    }

    cuadrado += lineaFuera;

    return cuadrado;
}

console.log(hacerCuadrado(1));
console.log(hacerCuadradoBucles(1));